packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -78,30 +85,62 @@ export class SCWSigner implements Signer {
     this.handleResponse(args, decrypted);
   }
 
-  async request(request: RequestArguments) {
+  // TODO: Properly type the return value
+  async request(request: RequestArguments): Promise<any> {
     if (this.accounts.length === 0) {
       switch (request.method) {
         case 'wallet_switchEthereumChain': {
           assertParamsChainId(request.params);
           this.chain.id = Number(request.params[0].chainId);
           return;
         }
-        case 'wallet_connect':
-        case 'wallet_sendCalls':
+        case 'wallet_connect': {
+          // Wait for the popup to be loaded before making async calls
+          await this.communicator.waitForPopupLoaded?.();
+          await initSubAccountConfig();
+          const modifiedRequest = injectRequestCapabilities(
+            request,
+            store.subAccountsConfig.get()?.capabilities ?? {}
+          );
+          return this.sendRequestToPopup(modifiedRequest);
+        }
+        case 'wallet_sendCalls': {
           return this.sendRequestToPopup(request);
-        default:
```

Comment:
We should likely keep this. It will cause an issue with the `eth_requestAccounts` in the main switch case. We will need to move the `eth_requestAccounts` into the no accounts section (this switch case)

Reply:
fixed. this also fixes the caching behavior that @fan-zhang-sv mentioned